pkg/resources/provisioning.cattle.io/v1/cluster/validator_test.go

```diff
@@ -131,3 +136,30 @@ func Test_isValidName(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateMachinePoolName(t *testing.T) {
```

Comment:
Everything else is LGTM, but could we add a few more test cases?

Reply:
Yep! Added a few more and changed to the struct approach like the test above.